swift-extensions/TrikotPublisherExtensions.swift

```diff
@@ -57,17 +65,25 @@ extension NSObject {
 
 extension NSObject {
     public func observe<V>(_ concretePublisher: ConcretePublisher<V>, toClosure closure: @escaping ((V) -> Void)) {
-        observe(cancellableManager: trikotInternalPublisherCancellableManager, concretePublisher: concretePublisher, toClosure: closure)
+        if let publisher = publisher as? JustPublisher<AnyObject> {
+            publisher.values.forEach { closure($0 as! V) }
+        } else {
+            observe(cancellableManager: trikotInternalPublisherCancellableManager, concretePublisher: concretePublisher, toClosure: closure)
+        }
     }
 
     public func observe<V>(cancellableManager: CancellableManager, concretePublisher: ConcretePublisher<V>, toClosure closure: @escaping ((V) -> Void)) {
-        PublisherExtensionsKt.subscribe(concretePublisher, cancellableManager: cancellableManager) {(value: Any?) in
-            if Thread.current.isMainThread {
-                closure(value as! V)
-            } else {
-                MrFreezeKt.freeze(objectToFreeze: value)
-                DispatchQueue.main.async {
+        if let publisher = publisher as? JustPublisher<AnyObject> {
```

Comment:
Ça sert à rien sur un concrete, le cast va jamais fonctionner, le concrete publisher est un container par dessus le publisher qu'il "hold" !

`class ConcretePublisher<T>(private val publisher: Publisher<T>) : Publisher<T> {
    override fun subscribe(s: Subscriber<in T>) {
        publisher.subscribe(s)
    }
}`